apply-inovelli-defaults/src/main.rs

```diff
@@ -33,9 +33,15 @@ async fn main() -> anyhow::Result<()> {
         .finish();
     tracing::subscriber::set_global_default(subscriber).expect("setting default subscriber failed");
     tracing::debug!(cmdline = ?args, "Starting");
-    let config_file = File::open(&args.config_file)?;
-    let config: Vec<config::ConfigClause> = serde_yaml::from_reader(config_file)
-        .with_context(|| format!("Parsing config file {:?}", &args.config_file))?;
+
+    let config_file = File::open(&args.config_file)
+        .with_context(|| format!("Opening config file {:?}", &args.config_file))?;
+    let mut config: serde_yaml::Value = serde_yaml::from_reader(config_file)
+        .with_context(|| format!("Loading config file {:?}, yaml may be invalid", &args.config_file))?;
+    config.apply_merge()
+        .with_context(|| format!("Merging config file {:?}", &args.config_file))?;
+    let config: Vec<config::ConfigClause> = serde_yaml::from_value(config)
+        .with_context(|| format!("Parsing config file {:?}, unexpected format", &args.config_file))?;
```

Comment:
Considering that the config-file loading is a bit more involved now, could you pull these out into a config-file loading function (let's call it `load_config`)? Then, adding the config file to the error message context would be `load_config(&args.config_file).with_context(|| format!("Loading config file {:?}", &args.config_file))?` and all the with_context calls you have now could just use static strings telling users what step along the config-loading process failed.

Reply:
Sure, makes a lot of sense. I pulled the logic into the config module and added a couple tests as well.